runtime/src/ancient_append_vecs.rs

```diff
@@ -37,26 +34,22 @@ impl<'a> AccountsToStore<'a> {
     /// available_bytes: how many bytes remain in the primary storage. Excess accounts will be directed to an overflow storage
     pub fn new(
         mut available_bytes: u64,
-        stored_accounts: &'a [&'a FoundStoredAccount<'a>],
+        accounts: &'a [&'a FoundStoredAccount<'a>],
         slot: Slot,
     ) -> Self {
-        let num_accounts = stored_accounts.len();
-        let mut accounts = Vec::with_capacity(num_accounts);
+        let num_accounts = accounts.len();
         // index of the first account that doesn't fit in the current append vec
         let mut index_first_item_overflow = num_accounts; // assume all fit
-        stored_accounts.iter().for_each(|account| {
+        for (i, account) in accounts.iter().enumerate() {
             let account_size = account.account.stored_size as u64;
             if available_bytes >= account_size {
                 available_bytes = available_bytes.saturating_sub(account_size);
             } else if index_first_item_overflow == num_accounts {
-                available_bytes = 0;
                 // the # of accounts we have so far seen is the most that will fit in the current ancient append vec
-                index_first_item_overflow = accounts.len();
+                index_first_item_overflow = i;
+                break;
```

Comment:
we can stop looking through accounts as soon as we find the one where we can no longer fit